cypress/integration/remote-storage/configurations/crud.spec.js

```diff
@@ -0,0 +1,127 @@
+      .expect([
+        Pane('Create configuration').exists(),
+        TextField('Remote storage name\n*').is({ focused: true, value: '' }),
+        Select('Provider name\n*').has({ value: '' }),
+        Button('Cancel').exists(),
+        Button('Save & close').exists(),
+      ])
+
+      .do([
+        TextField('Remote storage name\n*').fillIn(NAME),
+        Select('Provider name\n*').choose(PROVIDER),
+        Button('Save & close').click(),
+      ])
+      .expect([
+        Modal('Create configuration').exists(),
+        Modal('Create configuration').find(Button('Cancel')).exists(),
+        Modal('Create configuration').find(Button('Save')).exists(),
+      ])
+
+      .do(Button('Save').click())
+      .expect([
+        Modal('Create configuration').absent(),
+        Pane('Create configuration').absent(),
+        MultiColumnList('storages-list').exists(),
+        MultiColumnListCell(NAME).exists(),
+      ]);
+
+    // Read
+    cy.do(MultiColumnListCell(NAME).click())
+      .expect([
+        Pane(NAME).exists(),
+        Pane(NAME).find(Button('Actions')).exists(),
+        KeyValue('Remote storage name').has({ value: NAME }),
+        KeyValue('Provider name').has({ value: PROVIDER }),
+      ]);
+
+    // Update
+    cy.do(Button('Actions').click())
+      .expect(Button('Edit').exists())
+
+      .do(Button('Edit').click())
+      .expect([
+        Pane(`Edit ${NAME}`).exists(),
+        TextField('Remote storage name\n*').is({ focused: true, value: NAME }),
+        Select('Provider name\n*').has({ value: PROVIDER_CODE }),
+        Button('Cancel').exists(),
+        Button('Save & close').exists(),
+      ])
+
+      .do([
+        TextField('Remote storage name\n*').fillIn(NAME_EDITED),
+        Button('Save & close').click(),
+      ])
+      .expect([
+        Modal(`Edit ${NAME}`).exists(),
+        Modal(`Edit ${NAME}`).find(Button('Cancel')).exists(),
+        Modal(`Edit ${NAME}`).find(Button('Save')).exists(),
+      ])
+
+      .do(Button('Save').click())
+      .expect([
+        Modal(`Edit ${NAME}`).absent(),
+        Pane(`Edit ${NAME}`).absent(),
+        MultiColumnListCell(NAME).absent(),
+        MultiColumnListCell(NAME_EDITED).exists(),
+      ])
+
+      .do(MultiColumnListCell(NAME_EDITED).click())
+      .expect([
+        Pane(NAME_EDITED).exists(),
+        Pane(NAME_EDITED).find(KeyValue('Remote storage name')).has({ value: NAME_EDITED }),
+      ]);
+
+    // Delete
+    cy.do(Button('Actions').click())
+      .expect(Button('Delete').exists())
+
+      .do(Button('Delete').click())
+      .expect([
+        Modal(`Remove ${NAME_EDITED}`).exists(),
+        Modal(`Remove ${NAME_EDITED}`).find(Button('Cancel')).exists(),
+        Modal(`Remove ${NAME_EDITED}`).find(Button('Delete')).exists(),
+      ])
+
+      .do(Button('Delete').click())
+      .expect([
+        Modal(`Remove ${NAME_EDITED}`).absent(),
+        MultiColumnListCell(NAME_EDITED).absent(),
+      ]);
+  });
```

Comment:
optional, but you can probably get away without using the `NAME_EDITED` constant so much by using a matcher. Also, I've removed the redundant interactions.

```suggestion
    const modal = Modal(including('Remove '));
    cy.do(Button('Actions').click())
      .do(Button('Delete').click())
      .expect(modal.exists())
      .do(Button('Delete').click())
      .expect([
        modal.absent(),
        MultiColumnListCell(NAME_EDITED).absent(),
      ]);
  });
```

Reply:
@axelhunn It should because `Modal(including('Remove'))` is not a cypress command, it is just an expression that returns a data structure. In other words, `Modal()` is a pure function.

It isn't until you say `modal.exists()` or `modal.click()` that anything command-like happens. In fact, one of the suggestions that I was going to make, but forgot to was that you could actually create "save" and "delete" interactors for example at the very top of your file and then use them throughout the tests:

```js
const saveButton = Button('Save');
const deleteButton = Button('Delete');
```

You can now use these everywhere. So, for example, we could re-write the above as:

```js
    const modal = Modal(including('Remove '));
    cy.do([
      Button('Actions').click(),
      deleteButton.click()),
      modal.find(deleteButton.click())
    ])
      .expect([
        modal.absent(),
        MultiColumnListCell(NAME_EDITED).absent(),
      ]);
  });
```

Reply:
Thanks, it's a great thing!

Reply:
Will `.find()` API work this way, outside the `do` or `expect` command?

I mean can I do it like that:
```js
const pane = Pane(NAME);
const button = pane.find(Button('Actions'));

cy.do(button.click())
```

Reply:
It will indeed!